service/src/main/java/bio/terra/workspace/service/crl/CrlService.java

```diff
@@ -167,6 +168,35 @@ public boolean canReadBigQueryDataset(
     }
   }
 
+  /**
+   * Wrap the BigQuery dataset fetch in its own method. This allows unit tests to mock this service
+   * and generate an answer without actually touching BigQuery.
+   *
+   * @param bigQueryCow BigQuery client object, wrapped by CRL
+   * @param projectId Google project id where the dataset is
+   * @param datasetName name of the dataset
+   * @return the fetched Dataset object
+   */
+  public Dataset getBigQueryDataset(BigQueryCow bigQueryCow, String projectId, String datasetName)
+      throws IOException {
+    return bigQueryCow.datasets().get(projectId, datasetName).execute();
```

Comment:
I spent a while trying to get that to work, ultimately unsuccessfully. I hit two problems:
- I couldn't mock `execute` because Mockito didn't recognize the expected return value for some reason. It's supposed to return a Dataset, but Mockito wanted me to return an Optional. Not sure why -- maybe an inheritance problem?
- I couldn't mock a `Dataset` because it's a final class. There is an experimental feature in Mockito to enable this, but unclear if that will be supported long-term.